addons/base_vat/models/res_partner.py

```diff
@@ -363,3 +363,74 @@ def check_vat_al(self, vat):
             return stdnum.al.vat.is_valid(vat)
         except ImportError:
             return True
+
+    def check_vat_nl(self, vat):
+        """ Temporary Netherlands VAT validation to support the new format
+        introduced in January 2020, until upstream is fixed.
+        TODO: remove when fixed upstream"""
+
+        if not (len(vat) == 14):
+            return False
+
+        #checking the format
+        check_vat_nl_re = re.compile("(?:NL)?[0-9]{9}B[0-9]{2}")
+        match = check_vat_nl_re.match(vat)
+        if not match:
+            return False
+
+        char_to_int = {
+            'A' : 10,
+            'B' : 11,
+            'C' : 12,
+            'D' : 13,
+            'E' : 14,
+            'F' : 15,
+            'G' : 16,
+            'H' : 17,
+            'I' : 18,
+            'J' : 19,
+            'K' : 20,
+            'L' : 21,
+            'M' : 22,
+            'N' : 23,
+            'O' : 24,
+            'P' : 25,
+            'Q' : 26,
+            'R' : 27,
+            'S' : 28,
+            'T' : 29,
+            'U' : 30,
+            'V' : 31,
+            'W' : 32,
+            'X' : 33,
+            'Y' : 34,
+            'Z' : 35,
+            '+' : 36,
+            '*' : 37,
+        }
+
+        # removing the prefix
+        number = vat[2:]
+
+        # there are 2 possible checks
+        check_sole = False
+        check_company = False
+
+        # sole proprietary => mod97 checksum
+        check_val_sole = '2321'
+        for x in number:
+            if x.isdigit(): check_val_sole += x
+            else: check_val_sole += str(char_to_int[x])
+        check_sole = int(check_val_sole) % 97 == 1
+
+        # company => weighted mod11
+        company_sum = 0
+        counter = 0
+        for i in range(9, 1, -1):
+            company_sum += int(number[counter + 0 : counter + 1]) * i
+            counter += 1
+
+        check_val_company = 0 if (company_sum % 11) > 9 else (company_sum % 11)
+        check_company = check_val_company == int(vat[10:11])
```

Comment:
this should be a separated function _is_company_valid or something like that

Reply:
Maybe you can ispired yourself onto this : https://github.com/arthurdejong/python-stdnum/blob/087c668560148de86084b247c4dffafb19f5c3fe/stdnum/nl/bsn.py#L59
The checksum is smaller and you don't need to do the char_to_int